### ISSUE TITLE
Double Values, one invisible

### ISSUE THREAD

**Issue by repositoryofexcellence** [moroshko/react-autosuggest#444]
Hello. My question is, i want to display the value visible to the input, but want to pass the code by invisible to the ajax form too. Is that possible? My suggestion list is like this:

`const suggestions = [

    {
        code: 'SP',
        label: 'Sahinn Paradise',
    },
    {
        code: 'EDH',
        label: 'Elvin Deluxe Hotel',
    },
]`

**Comment by julian69** [comment]
I'm in the same situation and just now struggling to find a way to do it...

**Comment by moroshko** [comment]
Sorry, I don't understand the question.
Could you rephrase, please?

**Comment by julian69** [comment]
In my case, what I need is to display a value and send another one. I'm trying to do it by creating an input hidden, but I couldn't find yet a consistent way of handling that value (the hidden one). For instance, this what I'm trying at this point:
`getSuggestionValue(suggestion){

        if (this.props.inputHidden) {
            
            this.setState({
                hiddenValue: suggestion[this.props.inputHidden.propFromListAsInputValue]
            });
        }

        return suggestion[this.props.propFromListAsInputName];
    }`

Is there an existing approach to this situation?
Thanks!

**Comment by moroshko** [comment]
@julian69 Setting the state in `getSuggestionValue` is definitely the wrong approach. `getSuggestionValue` should be a pure function mapping a suggestion to its input value.

Autosuggest doesn't send anything.
Its purpose is to help you to fill in the input faster.
The return value of `getSuggestionValue()` will always become the input's value.
It's up to you how you display the suggestions (using `renderSuggestion`).
But, once you select a suggestion, `getSuggestionValue()` will determine the value of the input. You cannot change that.

I have a feeling that `react-autosuggest` might not be the best choice for your use case. Have a look at [react-select](https://github.com/JedWatson/react-select).

**Comment by matthieugayon** [comment]
it's a very common case though ... I guess smthg's possible with the suggestionIndex arg passed in the onSuggestionSelected hook
